enterprise/server/usage_service/usage_service.go

```diff
@@ -106,6 +98,20 @@ func (s *usageService) GetUsage(ctx context.Context, req *usagepb.GetUsageReques
 	return rsp, nil
 }
 
+func (s *usageService) GetUsage(ctx context.Context, req *usagepb.GetUsageRequest) (*usagepb.GetUsageResponse, error) {
+	groupID := req.GetRequestContext().GetGroupId()
+	if err := authutil.AuthorizeGroupAccess(ctx, s.env, groupID); err != nil {
+		return nil, err
+	}
```

Comment:
nit: the auth middleware is aware of the preferred group ID in the request context, so you could instead write `env.GetAuthenticator().GetAuthenticatedUser()` then `u.GetGroupID()`

Reply:
sure sure--this was just a move of what was there before, but happy to update it